### PR TITLE
Remove build and doc references to Git submodules

### PR DESCRIPTION
**Pull request by andyleejordan**
As of #7892, the PowerShell repository no longer uses Git submodules.
This is fantastic from a workflow standpoint, and so all the notes about
how to deal with submodules (and all the build steps explicitly
initializing and updating submodules) can be safely removed.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by andschwa** [comment]
Hi @TravisEz13!

Good job folks getting rid of submodules, that's fantastic 🎉

File: CHANGELOG.md [hunk 1]
```diff
@@ -48,6 +48,7 @@
 
 ### Code Cleanup
 
+- All references to Git submodules removed since there are no more.
```
Comment (adityapatwardhan) [inline CHANGELOG.md]: This should be added in the CHANGELOG for the next release. It will be automatically added when this PR gets merged.

**Review by adityapatwardhan — DISMISSED** [review]
Thanks for this PR. Just 1 comment about the CHANGELOG.

**Comment by adityapatwardhan** [comment]
@andschwa  Thanks for this PR!

**Review by anmenaga — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (andschwa) [reply CHANGELOG.md]: Fixed.

**Review by iSazonov — APPROVED** [review]

**Review by TravisEz13 — APPROVED** [review]

**Comment by TravisEz13** [comment]
Thanks for your contribution

**Pull request merged**